### DISCUSSION TITLE
Can we deploy page on our own server?

### DISCUSSION THREAD

**Discussion by chendwww** [mui/toolpad#2432] [Help and support] [answered]
Hello, I just start to use Toolpad. I wonder if we could deploy the pages we build by toolpad to our own server?

I read a doc on https://mui.com/toolpad/tutorials/render-deploy/.  but our server is on AWS we don't use render. so it would be possible to do it?  

If not, where can I find the file of the page I made? I want to check the file code and try to figure  it out by our team

Thanks

**Comment by oliviertassinari** [comment]
@chendwww Yes, you can. As of today, you need 1 GB of RAM and Node.js 18 or above.

cc @prakhargupta1 I think we need to create more deployment instructions, to at least explain at a high level how it works. For example https://next-site-4vixvljn1.vercel.sh/docs/deployment#nodejs-server. This sounds a lot more important than covering each possible target. I mean, Render is one of my ways to deploy a Node.js process.

**Reply by chendwww** [reply]
Thank you, any doc about deploy Toolpad to our own server?

**Reply by Janpot** [reply, marked as answer]
@chendwww We will add documentation on how to deploy on any server that has Node.js installed.
In short the process will be as follows:

> Make sure to add the following scripts to your `package.json`
>
> ```json
> "scripts": {
>   "dev": "toolpad dev"
>   "build": "toolpad build"
>   "start": "toolpad start"
> }
> ```
>
> - clone your project on your server
> - run `npm install`
> - run `npm run build`
> - run `npm run start`
>
> If you want to listen on a  specific port you can change the `start` script to:
>
> ```json
>   "start": "toolpad start -p 1234"
> ```

**Comment by bharatkashyap** [comment]
I think we can expand https://mui.com/toolpad/concepts/deployment/ with some more details to make deploying to any Node.js server clear (such as adding the paragraph Jan added in the discussion thread above)

**Reply by oliviertassinari** [reply]
What about putting Render under this page in the navigation? It feels strange to be disconnected.

**Reply by bharatkashyap** [reply]
I think we were going for the concept vs tutorial separation, with the Render page having a longer step-by-step instruction format, and the concept page being lean. I think a separate section scales well since we can potentially put all other deployment services under that section, without adding much to this concept page.

**Reply by prakhargupta1** [reply]
I don't think 'Deploying on Render' should be in the Tutorial section. Tutorials should be for Toolpad-specific write-ups. 

As we'll add instructions on how to deploy on other hosting providers, I think they should all be in a single place under Deployment. And since Deployment is not a Toolpad concept per se, it should be under How-to-guides.

**Reply by bharatkashyap** [reply]
> I don't think 'Deploying on Render' should be in the Tutorial section. Tutorials should be for Toolpad-specific write-ups.
> 
> As we'll add instructions on how to deploy on other hosting providers, I think they should all be in a single place under Deployment. And since Deployment is not a Toolpad concept per se, it should be under How-to-guides.

Fair enough, I'm on board with implementing this
